src/services/context/GlobalContext.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { createContext } from "react";
+import { iApiResponse } from "../api-manager/apiManagerInterfaces";
+import { iGetCategoryOptionsActionType } from "../../redux/types/book/bookType";
+import { Dispatch } from "redux";
+
+export interface iGlobalContext {
+  showLoader: boolean;
+  setShowLoader: React.Dispatch<boolean>;
+  globalDispatch: (
+    effect: (
+      dispatch: Dispatch<iGetCategoryOptionsActionType>
```

Comment:
dispatch will possibly have type of all the actions  so i think i will be something like  `bookActionTypes | userActionTypes`.